Pspcl/Pspcl.Web/Controllers/DeleteStockController.cs

```diff
@@ -23,15 +24,23 @@ public DeleteStockController(ApplicationDbContext _dbContext, IStockService stoc
 
         public IActionResult DeleteStock()
         {
-            try
+            if (!User.IsInRole("NonAdmin"))
```

Comment:
Wrong way to apply role check. Have some R&D about, how we decorate actions to apply role checks using authorize.